### ISSUE TITLE
SWRevealViewControllerSeguePushController with UITabBarContrller

### ISSUE THREAD

**Issue by user2233** [John-Lluch/SWRevealViewController#633]
I have a project working with the RevealController as initial VC, the "hamburger" slide-in sidebar as the rear VC and a UITabBar Controller as the frontVC displaying content.  Everything works fine up to that point.

In storyboard I have segues from menu buttons to the target View Controllers set  to SWRevealViewControllerSeguePushController as suggested int he documentation.  The target VCs are embedded in navigation controllers to show the toolbar items and before tapping you see the toolbar items in the toolbar. The problem is once you tap a button and segue to a target VC, items in the toolbar disappear.  If I change the segues to show, then toolbar items do show up on segue, however, the reveal functionality is lost and you cannot retrieve the sidebar menu.

Might there be a switch in the project to display the toolbars?  Or can anyone suggest a way to preserve access to the item icons in the toolbar after performing a SWRevealViewControllerSeguePushController segue?

**Comment by iDevelopper** [comment]
I have a sample for this:

[SWTabBarSwift.zip](https://github.com/John-Lluch/SWRevealViewController/files/548057/SWTabBarSwift.zip)

**Comment by user2233** [comment]
Great project.  Unfortunately for me, I have not yet learned Swift.  Looks like you are just using regular push controllers to the targets.  Could you explain approach  or what you modified to get it to work?

**Comment by iDevelopper** [comment]
I have an other one written in Objective-c:

[SWTabBar.zip](https://github.com/John-Lluch/SWRevealViewController/files/550407/SWTabBar.zip)

**Comment by user2233** [comment]
Patrick, great project.  I haven't learned Swift so can't immediately
figure out what you did.  Would you mind describing what you have done to
get it to work?

Looks like you are using "Show" Segues, not the RevealPush ones.  How are
you preserving the SWReveal functionality once you go to the regular menu?

**Comment by iDevelopper** [comment]
Did you watch the last project written in Objective-c? I don't use segue to push the front view controllers (but it is the same), I use programmatically (in MenuTableViewController.m, the rear view controller):

```
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    UITabBarController *tabBarController = (UITabBarController *)self.revealViewController.frontViewController;
    UINavigationController *navController = tabBarController.viewControllers[indexPath.row];
    [navController popToRootViewControllerAnimated:NO];
    tabBarController.selectedIndex = indexPath.row;
    [self.revealViewController pushFrontViewController:tabBarController animated:YES];
}
```

**Comment by user2233** [comment]
Yes, thanks so much.  Using the delegate method did the trick and tabs now showing on push.  Am still struggling with passing data, however.   myVC \* destVC = [navController.viewControllers objectAtIndex:0];
      destVC.data = @"Data Point"; is not doing it.

**Comment by iDevelopper** [comment]
This is not a SW issue, but:

The front view controller is a UITabBarController. The myVC is embedded in a UINavigationController who is embedded in the UITabBarController:

UITabBarController -> UINavigationController -> myVC, then

```
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    UITabBarController *tabBarController = (UITabBarController *)self.revealViewController.frontViewController;
    UINavigationController *navController = tabBarController.viewControllers[indexPath.row];
    UIViewController *vc = navController.topViewController;
    if ([vc isKindOfClass:[ViewController2 class]]) {
        ((ViewController2 *)vc).data = @"Data";
    }
    [navController popToRootViewControllerAnimated:NO];
    tabBarController.selectedIndex = indexPath.row;
    [self.revealViewController pushFrontViewController:tabBarController animated:YES];
}
```

[SWTabBar 2.zip](https://github.com/John-Lluch/SWRevealViewController/files/552347/SWTabBar.2.zip)

**Comment by user2233** [comment]
That works great for VC 2.  Can't get it to work for VC1.  Could this be because VC1 is already loaded would need to be reloaded?

**Comment by iDevelopper** [comment]
Subclass SWRevealViewController (RevealViewController in the sample), donc forget to set this class in the storyboard for SW and implement prepareForSegue:

[SWTabBar 3.zip](https://github.com/John-Lluch/SWRevealViewController/files/555027/SWTabBar.3.zip)

**Comment by user2233** [comment]
This lets you preset the data on load.  Still can't figure out how to push data dynamically from menu to the VC originally in stack.  For example, if you have a table of books and want the menu to list authors so the table would then show books by the selected author.

**Comment by iDevelopper** [comment]
You have just to keep a reference to your controller in a property or with a protocol. But I don't really understand, in didSelectRowAtIndexPath, it is done. Can I have a look at your storyboard?

**Comment by YouSour** [comment]
good but it need entry point to start on that view controller first how about i have other view controller i need to start it first , how can i fix it ?

**Comment by iDevelopper** [comment]
Insert another view controller as the entry point and instantiate SWRevelViewController from it when needed. What is the problem?

**Comment by YouSour** [comment]
ok , it work now , @iDevelopper , but when i have segue from table cell to UI tab bar controller item disappeared ?

**Comment by iDevelopper** [comment]
Can you attach your project here?

**Comment by YouSour** [comment]
here [](https://gitlab.com/Samnieng_tan/Home_Tutor) @iDevelopper

**Comment by YouSour** [comment]
[Home-tutor](https://gitlab.com/Samnieng_tan/Home_Tutor) here @iDevelopper

**Comment by iDevelopper** [comment]
404 not found! Upload it here...

**Comment by YouSour** [comment]
[Home_Tutor-master-89ba190111341eba76187c6e8cb5e44397098143.zip](https://github.com/John-Lluch/SWRevealViewController/files/1348642/Home_Tutor-master-89ba190111341eba76187c6e8cb5e44397098143.zip)
here @iDevelopper

**Comment by iDevelopper** [comment]
Ok, got it, but in your project there no segue between your table view and another controller?

**Comment by YouSour** [comment]
why don't you try add segue from table view to navigation controller of profile view controller you will see tab bar disappeared . @iDevelopper

**Comment by iDevelopper** [comment]
Ok, you have to push the tab bar controller (witch is the front view controller) and select his correct index. See LeftTableViewController.swift. I added also TutorSearchViewController and TutorNotificationsViewController.swift.

[Home_Tutor-master-89ba190111341eba76187c6e8cb5e44397098143.zip](https://github.com/John-Lluch/SWRevealViewController/files/1351407/Home_Tutor-master-89ba190111341eba76187c6e8cb5e44397098143.zip)

**Comment by YouSour** [comment]
wow , you 're good person @iDevelopper thank you so much , you save me :)
